apps/laboratory/tests/shared/validators/WalletValidator.ts

```diff
@@ -15,7 +15,7 @@ export class WalletValidator {
   }
 
   async expectDisconnected() {
-    await this.page.waitForTimeout(1000)
+    await this.page.waitForTimeout(2000)
     await this.page.reload()
     await this.gotoSessions.click()
     await expect(this.page.getByTestId('session-card')).not.toBeVisible()
```

Comment:
Any way to do this without an artificial delay?

Does `session-card` update dynamically? If so, you should be able to just wait for the condition without a timeout or reload at all?

If not, you might be able to use https://playwright.dev/docs/test-assertions#expectpoll or another polling mechanism.

Reply:
Yes this needs to be removed. background: I think this was related to the fact that using `ethers` the `await` on `disconnect` immediately resolves.

[I'll try your suggestion](https://github.com/WalletConnect/web3modal/issues/1582)